### PR TITLE
Update OpenTelemetry send data to Promscale

### LINKED ISSUE #925: Update Send data / OpenTelemetry to provide more details
We need to provide more details on the configuration options that are needed to send data from OpenTelemetry to Promscale.

Sending data via the OTel Collecto:  provide a complete configuration file that also includes the pipelines.
Point the OTel SDK directly to Promscale: explain they can use the environment variable OTEL_EXPORTER_OTLP_TRACES_ENDPOINT which is standard across all SDKs (if I'm not mistaken) on top of pointing users to the OTel docs. Also indicate that we only support gRPC. This means the SDK must support it or they will need to use the OTel Collector. For example the Ruby SDK at the moment only supports OTLP via HTTP.

### PR DESCRIPTION
**Pull request by VineethReddy02**
# Description

Add OpenTelemetry send data details. 

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)

# Links

Fixes #925

### CODE REVIEW

File: promscale/send-data/opentelemetry.md [hunk 1]
```diff
@@ -2,32 +2,69 @@
...
 parameter when you started the Promscale connector.
 
-Open the OpenTelemetry collector configuration file, and forward the OTLP traces to Promscale by adding or editing these lines:
+The below OpenTelemetry collector configuration file, configures 
+  * **Receivers**: Jaeger (gRPC, thrift http), OpenTelemetry line protool (gRPC, http)
+  * **Exporters**: OpenTelemetry line protocol configure the Promscale gRPC server to export the data to Promscale, queue_size to hold the data before dropping and `timeout` is timeout for write request to backend. 
+  * **Processors**: Batch processor is to batch, better compress the data and to control the number of outgoing connections, configured with `send_max_batch_size` and `timeout` to send data as per Promscale recommendation.
+  * **Service**: Pipeline is to configure to receive the data from Jaeger and OTLP, batch process them and to export the data in logs and to OTLP supported backend.
+
+With below configuration you can send the traces from Jaeger and OpenTelemetry applications to the collector and can be exported the ame to Promscale.
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:18]:
```suggestion
With this configuration you can send the traces from Jaeger and OpenTelemetry applications to the collector and export them to Promscale. 
```

File: promscale/send-data/opentelemetry.md [hunk 2]
```diff
@@ -2,32 +2,69 @@
...
 ```
 
-Replace `<connector-host>` and `<OTLP-port>` with the corresponding hostname and port. 
+Replace `<PROMSCALE_HOST>` and `<gRPC_PORT>` with the corresponding Promscale hostname and gRPC port. 
 For example, if you are running the OTLP Collector and the Promscale Connector on a 
 Kubernetes cluster the endpoint parameter would be similar to
 `endpoint: "promscale-connector.default.svc.cluster.local:9202"`
 
-# Configure the OpenTelemetry instrumentation
+## Send the data using OpenTelemetry instrumentation
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:60]:
```suggestion
## Send data using OpenTelemetry instrumentation
```

File: promscale/send-data/opentelemetry.md [hunk 3]
```diff
@@ -2,32 +2,69 @@
...
 We recommend that you use the OpenTelemetry Collector to export data to an 
-observability backend in production. However, you can also send data from 
+observability backend in production as it offers batching, queued retries and many other functionalities by configuring the processors. However, you can also send data from
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:62]:
```suggestion
observability backend in production as it offers batching, queued retries, and other functions by configuring the processors. However, you can also send data from 
```

File: promscale/send-data/opentelemetry.md [hunk 4]
```diff
@@ -2,32 +2,69 @@
...
 
-# Configure the OpenTelemetry instrumentation
+## Send the data using OpenTelemetry instrumentation
 We recommend that you use the OpenTelemetry Collector to export data to an 
-observability backend in production. However, you can also send data from 
+observability backend in production as it offers batching, queued retries and many other functionalities by configuring the processors. However, you can also send data from 
 OpenTelemetry instrumentation libraries and SDKs directly to Promscale using 
 OTLP.
 
+With Opentelemetry SDKs you can configure the exporter backend using `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` environment variable on your applications end.
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:66]:
```suggestion
With Opentelemetry SDKs you can configure the exporter backend using the `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` environment variable on your applications' end. 
```

File: promscale/send-data/opentelemetry.md [hunk 5]
```diff
@@ -5,17 +5,17 @@ instrumentation libraries or the OpenTelemetry Collector to send traces to
...
+You can configure the OpenTelemetry collector to forward traces, metrics to Promscale. Promscale listens
 to OTLP data on the port you specified with the `otlp-grpc-server-listen-address`
-parameter when you started the Promscale connector.
+parameter this defaults to port `:9202` and Promscale listen to Prometheus data on the port you specified with the `ee` this defaults to port `:9201` when you started the Promscale connector.
 
 The below OpenTelemetry collector configuration file, configures 
   * **Receivers**: Jaeger (gRPC, thrift http), OpenTelemetry line protool (gRPC, http)
-  * **Exporters**: OpenTelemetry line protocol configure the Promscale gRPC server to export the data to Promscale, queue_size to hold the data before dropping and `timeout` is timeout for write request to backend. 
+  * **Exporters**: OpenTelemetry line protocol configure the Promscale gRPC server to export the data to Promscale, queue_size to hold the data before dropping and `timeout` is timeout for write request to backend, Prometheus Remote Write is to export the OpenTelemetry line protocol metrics to Prometheus remote storage backend configure the Promscale http endpoint to ingest the metrics.
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:14]:
```suggestion
  * **Exporters**: OpenTelemetry line protocol to configure the Promscale gRPC server to export the data to Promscale. Use `queue_size` to hold the data before dropping, and `timeout` for the write request to the backend. Prometheus remote write exports the OpenTelemetry line protocol metrics to the Prometheus remote storage backend. Configure the Promscale HTTP endpoint to ingest the metrics. 
```

Not 100% sure on this edit, please check.

File: promscale/send-data/opentelemetry.md [hunk 6]
```diff
@@ -5,17 +5,17 @@ instrumentation libraries or the OpenTelemetry Collector to send traces to
...
-  * **Exporters**: OpenTelemetry line protocol configure the Promscale gRPC server to export the data to Promscale, queue_size to hold the data before dropping and `timeout` is timeout for write request to backend. 
+  * **Exporters**: OpenTelemetry line protocol configure the Promscale gRPC server to export the data to Promscale, queue_size to hold the data before dropping and `timeout` is timeout for write request to backend, Prometheus Remote Write is to export the OpenTelemetry line protocol metrics to Prometheus remote storage backend configure the Promscale http endpoint to ingest the metrics. 
   * **Processors**: Batch processor is to batch, better compress the data and to control the number of outgoing connections, configured with `send_max_batch_size` and `timeout` to send data as per Promscale recommendation.
-  * **Service**: Pipeline is to configure to receive the data from Jaeger and OTLP, batch process them and to export the data in logs and to OTLP supported backend.
+  * **Service**: Pipeline is to configure to receive the data from Jaeger and OTLP (traces, metrics), batch process them and to export the data in logs and to OTLP, Prometheus supported backends.
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:16]:
```suggestion
  * **Service**: Pipeline to configure to receive the traces and metrics from Jaeger and OTLP, batch process them, and export the data in logs to OTLP and Prometheus-supported backends.
```

File: promscale/send-data/opentelemetry.md [hunk 7]
```diff
@@ -50,20 +55,24 @@ service:
...
 For example, if you are running the OTLP Collector and the Promscale Connector on a 
 Kubernetes cluster the endpoint parameter would be similar to
-`endpoint: "promscale-connector.default.svc.cluster.local:9202"`
+`endpoint: "promscale-connector.default.svc.cluster.local:<PORT>"`
 
-## Send the data using OpenTelemetry instrumentation
+## Send data using OpenTelemetry instrumentation
 We recommend that you use the OpenTelemetry Collector to export data to an 
-observability backend in production as it offers batching, queued retries and many other functionalities by configuring the processors. However, you can also send data from 
+observability backend in production as it offers batching, queued retries, and many other functions by configuring the processors. However, you can also send data from
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:71]:
```suggestion
observability backend in production as it offers batching, queued retries, and many other functions by configuring the processors. However, you can also send data from 
```

We need to add guidance on hyphenation for words like backend to our word usage guide. For `backend` IBM says to hyphenate, Google says one word 🤷‍♀️

**Review by Loquacity — CHANGES_REQUESTED** [review]
I didn't mention it inline, but the whole document needs reflowing for line length, too.

**Review by Loquacity — CHANGES_REQUESTED** [review]
Still needs reflowing for line length.

**Comment by Rajakavitha1** [comment]
Hi @VineethReddy02 !!!! installed using [helm](https://github.com/open-telemetry/opentelemetry-helm-charts). Validated the configurations.

**Review by Rajakavitha1 — APPROVED** [review]

File: promscale/send-data/opentelemetry.md [hunk 8]
```diff
@@ -1,34 +1,106 @@
...
+      http:
+
 exporters:
+  logging:
   otlp:
-    endpoint: "<connector-host>:<OTLP-port>"
-    insecure: true
+    endpoint: "<PROMSCALE_HOST>:<gRPC_PORT>"
+    tls:
+      insecure: true
```
[the remark this answers is not in the document]
Comment (VineethReddy02) [reply promscale/send-data/opentelemetry.md:57]: Works as a starting point `insecure` should be fine, using TLS adds additional friction for the users getting started as it needs configuring certs both on the collector and Promscale end.

We can re-iterate over this based on incoming requests from the users on TLS.

File: promscale/send-data/opentelemetry.md [hunk 9]
```diff
@@ -1,34 +1,106 @@
...
+  batch:
+    send_batch_size: 4000
+    send_batch_max_size: 4000
+    timeout: 10s
+
+service:
+  pipelines:
+    traces:
+      receivers: [otlp]
+      exporters: [logging, otlp]
```
[the remark this answers is not in the document]
Comment (VineethReddy02) [reply promscale/send-data/opentelemetry.md:77]: logging exporter logs the state that spans are being ingested. It doesn't log the content of the trace it's optional. I will drop as it as no influence on the settings.

File: promscale/send-data/opentelemetry.md [hunk 10]
```diff
@@ -1,34 +1,106 @@
...
-Promscale natively supports the OpenTelemetry Line Protocol (OTLP) for 
-traces. You can use any of the OpenTelemetry client SDKs,
-instrumentation libraries or the OpenTelemetry Collector to send traces to
-Promscale using OTLP. Currently, only gRPC is supported.
+Promscale natively supports the OpenTelemetry Line Protocol (OTLP) for traces
+and Prometheus remote write protocol for OpenTelemetry metrics. You can use any
+of the OpenTelemetry client SDKs, instrumentation libraries, or the
+OpenTelemetry Collector to send traces and metrics to Promscale using OTLP.
+Currently, Promscale supports only **gRPC** for traces. The OpenTelemetry
+metrics are converted into Prometheus remote write format in OpenTelemetry
```
[the remark this answers is not in the document]
Comment (Rajakavitha1) [reply promscale/send-data/opentelemetry.md]: rephrased as: You can send metrics to Promscale
**HTTP** endpoint using Prometheus remote write protocol with the OpenTelemetry
Collector. OpenTelemetry Collector converts OTLP metrics to Prometheus remote
write protocol metrics.

File: promscale/send-data/opentelemetry.md [hunk 11]
```diff
@@ -8,21 +8,22 @@ libraries, or the OpenTelemetry Collector. Currently, Promscale only supports
 Collector. OpenTelemetry Collector converts OTLP metrics to Prometheus remote
 write protocol metrics.
 
+## Configure Promscale Connector
+Promscale listens to OTLP traces on the port you specify in the
+`tracing.otlp.server-address` parameter, the default port is `9202`. Promscale
+listens to Prometheus metrics on the port you specify in the
+`web.listen-address` parameter, the default port is `9201` when you start the
+Promscale Connector. For more information about configuring Promscale Connector,
+see [Promscale CLI reference][promscale-cli].
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md]:
```suggestion
see the [Promscale CLI reference][promscale-cli].
```

File: promscale/send-data/opentelemetry.md [hunk 12]
```diff
@@ -38,9 +39,15 @@ Set the following in OpenTelemetry Collector configuration file:
...
     on the settings in the `receivers`, `processors`, `exporters`, and
+<<<<<<< HEAD
+    `extensions` sections. Pipelines to receive the traces and metrics from OTLP,
+    batch process them, and export the data in logs and OTLP to backends
+    supported in Prometheus.
+=======
     `extensions` sections. Pipeline to receive the traces and metrics from OTLP,
     batch process them, and export the data to OTLP gRPC, Prometheus remote-write
     supported backends.
+>>>>>>> 48d6b2516b0d63b476dc3049714385792ba3e723
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md]: Merge conflict markers here.

**Review by Loquacity — CHANGES_REQUESTED** [review]

File: promscale/send-data/opentelemetry.md (hunk 12 above)
Comment (VineethReddy02) [reply promscale/send-data/opentelemetry.md]: These merge conflict markers don't exist anymore.

File: promscale/send-data/opentelemetry.md [hunk 13]
```diff
@@ -95,22 +80,15 @@ Where:
...
-endpoint by configuring the exporter backend. Set the  exporter backed using the
-`OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` environment variable on your application.
-Ensure that the SDK you are using on the application supports gRPC. If the SDK
-does not support gRPC then use the OpenTelmetry Collector. Use the OTLP `http
-receiver` and OTLP `gRPC exporter` collector configuration to convert OTLP http
-to gRPC.   
 
 The specifics of the configuration are different for each SDK and library. For
-more information, see[OpenTelemetry documentation][otel-docs].
+more information, see [OpenTelemetry documentation][otel-docs].
```
Comment (Loquacity) [inline promscale/send-data/opentelemetry.md:91]:
```suggestion
more information, see the [OpenTelemetry documentation][otel-docs].
```

**Review by Loquacity — APPROVED** [review]

**Pull request merged**